### PR TITLE
SQ-56/Loginnnnnnnn (no UI yet)

### PR DESCRIPTION
**Pull request by rock3r**
WE HAVE GOOGLE SIGN-IN!

The favourites now require the Google Sign-in as it simplifies all edge cases (we don't have the ability to do a migration yet, maybe @fourlastor will do it later on but we'll see). There is no UI work so far as we have no designs... I'll come up with something if @DigitalPencils can't.

Most of the code was done by @fourlastor to which I provided spiritual support and food.

h/t to @frapontillo for helping out with some nasty Firebase fuckups (not cancelling callbacks when you sign out? Nice.)

signin

### CODE REVIEW

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.java [hunk 1]
```diff
@@ -77,11 +79,13 @@ public void onSpeakerClicked(Speaker speaker) {
...
             public void onFavoriteClick() {
-                if (event.favorited()) {
-                    subscriptions.add(service.removeFavorite(event.id()).subscribe());
-                } else {
-                    subscriptions.add(service.favorite(event.id()).subscribe());
-                }
+
+                subscriptions.add(service.toggleFavorite(event).subscribe(result -> {
+                    if (result == FavoriteResult.MUST_AUTHENTICATE) {
+                        Toast.makeText(EventDetailsActivity.this, "You need to sign in for this", Toast.LENGTH_SHORT).show();
```
Comment (tiwiz) [inline app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.java:85]: This one is gonna change when we do the UI?

File: app/src/main/java/net/squanchy/signin/SignInActivity.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        showProgressDialog();
+
+        service.signInWithGoogle(account)
+                .subscribeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::finish);
+    }
+
+    private void showSignInFailedError() {
+        hideProgressDialog();
+        Toast.makeText(this, "Google Play Services error.", Toast.LENGTH_SHORT).show();
```
Comment (tiwiz) [inline app/src/main/java/net/squanchy/signin/SignInActivity.java:76]: This one will go away as well as the following others, I guess?

**Review by tiwiz — APPROVED** [review]

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.java:85]: Of course

File: app/src/main/java/net/squanchy/signin/SignInActivity.java (hunk 2 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/signin/SignInActivity.java:76]: Yep

**Pull request merged**